core/src/test/java/org/infinispan/dataconversion/DataConversionTest.java

```diff
@@ -174,7 +173,7 @@ public void call() {
             Cache<String, Person> cache = cm.getCache();
             cm.getClassAllowList().addClasses(Person.class);
             // Obtain cache with custom valueEncoder
-            Cache storeMarshalled = cache.getAdvancedCache().withEncoding(JavaSerializationEncoder.class);
+            Cache storeMarshalled = cache.getAdvancedCache();
```

Comment:
Should we use `withMediaType` ? Also should we have a `withMediaType(MediaType)`  method? I noticed we only have the two argument one.

Reply:
> Should we use withMediaType

The test mechanics changed :)

> should we have a withMediaType(MediaType) method?

I can add it

Reply:
I think the point is that the values need to be transcoded to `java/x-java-object` anyway, but I agree that it may be more relevant (and more similar to the initial version of the test) if `storeMarshalled` had used media type `application/x-java-serialized-object`.

Reply:
> > should we have a withMediaType(MediaType) method?
> 
> I can add it

You can add it for completeness, but IMO it doesn't make a lot of sense. The keys and values are usually going to have the same media type only when using a marshaller's media type, and then changing the media type doesn't make a lot of sense (except maybe protostream <-> JSON?)